src/ota.rs

```diff
@@ -325,7 +335,7 @@ impl EspOta {
     }
 
     fn check_read(&self) -> Result<(), EspError> {
-        if self.0.update_partition.is_null() {
```

Comment:
This section was previously a bit problematic, as `esp_ota_get_next_update_partition` was capable of returning null pointers in case the a update partition couldn't be found.

Keeping this in mind, working with an `Option<EspOtaUpdate>` seems like the cleaner solution anyway, as the default null pointer holds a bit of a special meaning

Reply:
I think - and as per above - we need a much much simpler change - which is - if `esp_ota_get_next_update_partition` is returning a null ptr, `initiate_update` should just fail returning an `EspError`.

I'm unsure as to what is the special meaning of a default null pointer, if you could elaborate?